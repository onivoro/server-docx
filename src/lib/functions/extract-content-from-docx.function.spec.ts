import { execPromise } from '@onivoro/server-process';
import { extractContentFromDocx } from './extract-content-from-docx.function';
import { getFilePaths } from './get-file-paths.function';

describe('extractContentFromDocx', () => {
    it('extracts the text from a docx file on disk', async () => {
        const filename = 'test.docx';
        const { inflated } = getFilePaths(filename);
        expect(await extractContentFromDocx(filename)).toMatchSnapshot();
        await execPromise(`rm -rf ${inflated}`);
    });
});