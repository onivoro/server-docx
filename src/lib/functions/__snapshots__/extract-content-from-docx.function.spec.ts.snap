// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractContentFromDocx extracts the text from a docx file on disk 1`] = `
Object {
  "content": "﻿<?xml version=\\"1.0\\" encoding=\\"utf-8\\" standalone=\\"yes\\"?><w:document xmlns:wpc=\\"http://schemas.microsoft.com/office/word/2010/wordprocessingCanvas\\" xmlns:mc=\\"http://schemas.openxmlformats.org/markup-compatibility/2006\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\" xmlns:r=\\"http://schemas.openxmlformats.org/officeDocument/2006/relationships\\" xmlns:m=\\"http://schemas.openxmlformats.org/officeDocument/2006/math\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:wp14=\\"http://schemas.microsoft.com/office/word/2010/wordprocessingDrawing\\" xmlns:wp=\\"http://schemas.openxmlformats.org/drawingml/2006/wordprocessingDrawing\\" xmlns:w10=\\"urn:schemas-microsoft-com:office:word\\" xmlns:w=\\"http://schemas.openxmlformats.org/wordprocessingml/2006/main\\" xmlns:w14=\\"http://schemas.microsoft.com/office/word/2010/wordml\\" xmlns:w15=\\"http://schemas.microsoft.com/office/word/2012/wordml\\" xmlns:wpg=\\"http://schemas.microsoft.com/office/word/2010/wordprocessingGroup\\" xmlns:wpi=\\"http://schemas.microsoft.com/office/word/2010/wordprocessingInk\\" xmlns:wne=\\"http://schemas.microsoft.com/office/word/2006/wordml\\" xmlns:wps=\\"http://schemas.microsoft.com/office/word/2010/wordprocessingShape\\" xmlns:w16se=\\"http://schemas.microsoft.com/office/word/2015/wordml/symex\\" xmlns:w16cid=\\"http://schemas.microsoft.com/office/word/2016/wordml/cid\\" xmlns:w16=\\"http://schemas.microsoft.com/office/word/2018/wordml\\" xmlns:w16cex=\\"http://schemas.microsoft.com/office/word/2018/wordml/cex\\" xmlns:w16sdtdh=\\"http://schemas.microsoft.com/office/word/2020/wordml/sdtdatahash\\" mc:Ignorable=\\"w14 w15 wp14 w16se w16cid w16 w16cex w16sdtdh\\"><w:body><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"251AA360\\" w14:textId=\\"47C7EA3F\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t>{{DATE}}</w:t></w:r></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"5DD5AE4C\\" w14:textId=\\"6EF78B69\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"66CBAD96\\" w14:textId=\\"2164E997\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t>Dear Spouse of {{RESIDENT NAME}},</w:t></w:r></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"39C032E1\\" w14:textId=\\"5598A303\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"088C7D84\\" w14:textId=\\"3146FA2C\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t>I’m</w:t></w:r><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t xml:space=\\"preserve\\"> worried about the estate of </w:t></w:r><w:r w:rsidRPr=\\"7C2BB9B4\\" w:rsidR=\\"7C2BB9B4\\"><w:rPr><w:b w:val=\\"1\\" /><w:bCs w:val=\\"1\\" /><w:color w:val=\\"4C94D8\\" w:themeColor=\\"text2\\" w:themeTint=\\"80\\" w:themeShade=\\"FF\\" /></w:rPr><w:t>{{RESIDENT NAME}}</w:t></w:r><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t xml:space=\\"preserve\\">. {{RESIDENT SUBJECT PRONOUN}} estate is small. </w:t></w:r></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"3B7CD81D\\" w14:textId=\\"6CB02239\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"5D27FB86\\" w14:textId=\\"56D03B68\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t>{{RESIDENT SUBJECT PRONOUN}} owes {{LITERAL AMOUNT OWED}}.</w:t></w:r></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"4ECB890C\\" w14:textId=\\"7E7919BB\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"4468EF65\\" w14:textId=\\"7F4943EB\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t>Sincerely,</w:t></w:r></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"0577469A\\" w14:textId=\\"7D74B19A\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr><w:r w:rsidR=\\"7C2BB9B4\\"><w:rPr /><w:t>{{ATTORNEY NAME}}</w:t></w:r></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"2DF50FA4\\" w14:textId=\\"4003804C\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"19EC6EBF\\" w14:textId=\\"5CD7DC0D\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"0DFDAA24\\" w14:textId=\\"26CDE74C\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:p w:rsidR=\\"7C2BB9B4\\" w:rsidP=\\"7C2BB9B4\\" w:rsidRDefault=\\"7C2BB9B4\\" w14:paraId=\\"698E9C85\\" w14:textId=\\"20A8F6BB\\"><w:pPr><w:pStyle w:val=\\"Normal\\" /></w:pPr></w:p><w:sectPr><w:pgSz w:w=\\"12240\\" w:h=\\"15840\\" w:orient=\\"portrait\\" /><w:pgMar w:top=\\"1440\\" w:right=\\"1440\\" w:bottom=\\"1440\\" w:left=\\"1440\\" w:header=\\"720\\" w:footer=\\"720\\" w:gutter=\\"0\\" /><w:cols w:space=\\"720\\" /><w:docGrid w:linePitch=\\"360\\" /></w:sectPr></w:body></w:document>",
  "contentPath": "test/inflated/word/document.xml",
  "inflated": "test/inflated",
  "inputFilePath": "test/test.docx",
  "outputFileName": "test-modified.docx",
  "outputFilePath": "test/test-modified.docx",
  "root": "test",
}
`;
